src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/FileSignInfoComparer.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.Maestro.Tasks
+{
+    /// <summary>
+    ///     Compares FileSignInfo objects based on file and certificate name
+    /// </summary>
+    public class FileSignInfoComparer : IEqualityComparer<FileSignInfo>
+    {
+        public bool Equals(FileSignInfo x, FileSignInfo y)
+        {
+            return x.CertificateName.Equals(y.CertificateName, StringComparison.OrdinalIgnoreCase) &&
+                x.File.Equals(y.File, StringComparison.OrdinalIgnoreCase);
```

Comment:
This file is a sub-path?

Reply:
Yeah it is just the file name with no path at all. `<FileSignInfo File="YamlDotNet.dll" CertificateName="3PartySHA2" />`